src/main/java/uk/gov/justice/hmpps/prison/service/AdjudicationsService.java

```diff
@@ -530,15 +530,45 @@ private List<Sanction> transform(List<OicSanction> oicSanctions) {
     public List<Sanction> updateOicSanctions(
         final Long adjudicationNumber,
         final List<OicSanctionRequest> oicSanctionRequests) {
+        List<Sanction> updatedSanctions = Lists.newArrayList();
 
         var result = validateOicSanction(adjudicationNumber);
 
-        Long nextSanctionSeq = oicSanctionRepository.getNextSanctionSeq(result.offenderBookId());
+        List<OicSanction> existingOicSanctionsToDeleteAndRecreate = new ArrayList<>(oicSanctionRepository.findByOicHearingId(result.oicHearingId()));
 
-        List<OicSanction> exitingOicSanctions = oicSanctionRepository.findByOicHearingId(result.oicHearingId());
-        oicSanctionRepository.deleteAll(exitingOicSanctions);
+        var existingAdaSanctions = existingOicSanctionsToDeleteAndRecreate.stream().filter(
+            sanction -> Objects.nonNull(sanction.getOicSanctionCode()) && sanction.getOicSanctionCode().equals(OicSanctionCode.ADA)).toList();
+
+       if (!existingAdaSanctions.isEmpty()) {
+            for (var adaSanction : existingAdaSanctions) {
+                var optionalLinkedSanctionForThisBookId = oicSanctionRepository.findById(new OicSanction.PK(adaSanction.getOffenderBookId(), adaSanction.getSanctionSeq()));
+
+                optionalLinkedSanctionForThisBookId.ifPresent(linkedSanctionForThisBookId -> {
+                    if (Objects.equals(adaSanction.getStatus(), linkedSanctionForThisBookId.getStatus())) {
+                        var sanctionDays = oicSanctionRequests.get(0).getSanctionDays();
+                        linkedSanctionForThisBookId.setSanctionDays(sanctionDays);
+                        oicSanctionRepository.save(linkedSanctionForThisBookId);
+
+                        adaSanction.setSanctionDays(sanctionDays);
+                        oicSanctionRepository.save(adaSanction);
+                        existingOicSanctionsToDeleteAndRecreate.remove(adaSanction);
+
+                        updatedSanctions.addAll(toDto(List.of(adaSanction, linkedSanctionForThisBookId)));
+                    }
+                });
+            }
+        }
+
+       if (!existingOicSanctionsToDeleteAndRecreate.isEmpty()) {
```

Comment:
In the scenario where we've updated an existing `adaSanction` then we remove it from existingOicSanctionsToDeleteAndRecreate - [line 554](https://github.com/ministryofjustice/prison-api/pull/1697/files#diff-dadab9a8f82365cf5227ec804663b0d13e0c907b45bc0c110b93e84f21bc91b1R554) - and the collection is empty then we don't need to call `saveToDto`